src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -67,17 +67,20 @@ internal static class PooledDictionaryIgnoringNullableModifiersForReferenceTypes
     }
 
     // Method type inference can fail, but we still might have some best guesses. 
-    internal struct MethodTypeInferenceResult
+    internal readonly struct MethodTypeInferenceResult
     {
         public readonly ImmutableArray<TypeWithAnnotations> InferredTypeArguments;
+        public readonly bool InferredFromFunctionType;
```

Comment:
nit: Consider adding a comment here and in `MemberResolutionResult`
nit: Consider aligning names between those two structures (`TypeArgumentsFromFunctionType` here?)